sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuParquetScan.scala

```diff
@@ -105,6 +121,12 @@ object GpuParquetScanBase {
         s"${RapidsConf.ENABLE_PARQUET_READ} to true")
     }
 
+    if (meta.conf.isParquetSmallFilesEnabled && sparkSession.conf
+      .getOption("spark.sql.parquet.mergeSchema").exists(_.toBoolean)) {
+      meta.willNotWorkOnGpu("mergeSchema is not supported yet with" +
+      s" the small file optimization, please disable ${ENABLE_SMALL_FILES_PARQUET.key}")
```

Comment:
Making the small file optimization a per-scan-exec constructor argument also allows us to avoid disabling GPU support entirely here.

Reply:
yep, changed